### PR TITLE
Implementing Dynamic Form Navigation Based on Spouse Information Changes

### PR DESCRIPTION
**Pull request by amponce**
## Summary
We developed a solution to dynamically control the navigation flow of our form based on changes in spouse information. This was necessary for situations where users might edit their spouse details on the review and submit page, triggering the need to show a transition page before proceeding.

### Solution:

1. Local State Tracking: Utilized the existing `useDetectFieldChanges `hook to monitor changes in spouse-related fields within the form. When a change in spouse details is detected, a local state flag (`spouseChanged`) is set to true.
2. Global State Manager: Created a simple global state manager to hold and update this state. This manager is accessible both within React components and outside in the form config, bridging the gap created by the inability to modify Redux states. The checkGlobalState.js can be used to monitor everything from spouse changes to income, expense and assets updates during the session. We might want to consider local storage as an option if we want the data to persist long than the editing session.
3. Form Config Dependency: Updated the form configuration to include a new dependency condition. This condition checks the global state manager to determine whether the spouse details have changed (`globalState.spouseChanged`). If so, and if the other conditions (`formData.questions.isMarried`, ` !formData.reviewNavigation` and `formData['view:streamlinedWaiver']`) are met, the form dynamically renders the spouse transition page.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#69219

## Testing done

- The page did not exist before the change was implemented
-  To test you'll need to change update the financial-status-report.scss 
```css
.form-review-panel-page-header-row {
  va-button,
  button {
    display: block;
  }
}
```

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| NEW   | 
| ------- |
| Mobile  | 
image 
| Desktop |   
image
| Spouse Text Update|     
image

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Comment by amponce** [comment]
https://github.com/department-of-veterans-affairs/vets-website/assets/3916436/3596cb4d-65ba-4138-ae7a-e7716acedefc

Functionality Clip

File: src/applications/financial-status-report/components/shared/StreamlinedExplainer.jsx [hunk 1]
```diff
@@ -9,36 +9,15 @@ const StreamlinedExplainer = ({
...
-      </fieldset>
-    </form>
+    <ExplainerComponent
+      headline="You can skip questions on this form"
+      strongMessage="Based on your responses so far, you’re tentatively eligible for debt relief. We don’t need to ask you any more questions."
+      detailMessage="After you submit your request, we’ll mail you a letter with more details."
+      contentBeforeButtons={contentBeforeButtons}
+      goBack={goBack}
+      goForward={goForward}
+      contentAfterButtons={contentAfterButtons}
```
Comment (amponce) [inline src/applications/financial-status-report/components/shared/StreamlinedExplainer.jsx:19]: These might not be needed in the component

File: src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <ExplainerComponent
+      headline="You added a spouse"
+      strongMessage="You will now be asked additional questions about your spouse’s
+      income and employment."
+      detailMessage="  After you answer these questions, you can continue back to the
+      review page."
+      contentBeforeButtons={contentBeforeButtons}
+      goBack={goBack}
+      goForward={goForward}
+      contentAfterButtons={contentAfterButtons}
```
Comment (amponce) [inline src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx:21]: May not be needed

File: src/applications/financial-status-report/components/shared/StreamlinedExplainer.jsx (hunk 1 above)
Comment (amponce) [reply src/applications/financial-status-report/components/shared/StreamlinedExplainer.jsx:19]: > ```js
contentBeforeButtons={contentBeforeButtons}
goBack={goBack}
goForward={goForward}
contentAfterButtons={contentAfterButtons}
```

File: src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  contentBeforeButtons,
+  contentAfterButtons,
+  goBack,
+  goForward,
+}) => {
+  return (
+    <ExplainerComponent
+      headline="You added a spouse"
+      strongMessage="You will now be asked additional questions about your spouse’s
+      income and employment."
```
Comment (josephrlee) [inline src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx:15]: Add these sentences here: 

`Your spouse is considered a dependent. On the dependents page, include your spouse as a dependent.`

**Review by josephrlee — CHANGES_REQUESTED** [review]
Just added a comment to add the updated sentences about the spouse dependent on the transition page.

File: src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx (hunk 3 above)
Comment (amponce) [reply src/applications/financial-status-report/components/householdIncome/SpouseTransitionExplainer.jsx:15]: Done!

**Review by Dr-Pongo — CHANGES_REQUESTED** [review]
Dropping some notes for future reference, if the spouse changes from true to false and they have employment history, after the veteran additional income section the next page is `/spouse-employment-work-dates` so we'll need to look at the `depends` for that page. After that page it continues to the spouse employment summary page. 

This kind of brings up an additional issue we need to investigate to make sure that when we switch a radio button from yes to no, we should clear out the relevant data, so if spouse goes yes to no, we remove all relevant spouse data (employment history, additional income, benefits). Same should go for stuff like recreational vehicles, vehicles, credit card bills, etc. 

I'm still in the midst of testing, but wanted to drop this in here! I think the spouse removal should be addressed in this set of changes, but the others likely warrant their own tickets to prevent bloat.

File: src/applications/financial-status-report/components/household/DependentCount.jsx [hunk 4]
```diff
@@ -28,7 +29,8 @@ const DependentCount = ({
 
   const [error, setError] = useState(null);
   const [dependents, setDependents] = useState(hasDependents);
-
+  // Hook will handle the logic based on isMarried.
+  useClearSpouseData(isMarried, data, setFormData);
```
Comment (amponce) [inline src/applications/financial-status-report/components/household/DependentCount.jsx:33]: Added check for spouse data here

**Comment by amponce** [comment]
> Dropping some notes for future reference, if the spouse changes from true to false and they have employment history, after the veteran additional income section the next page is `/spouse-employment-work-dates` so we'll need to look at the `depends` for that page. After that page it continues to the spouse employment summary page.
> 
> This kind of brings up an additional issue we need to investigate to make sure that when we switch a radio button from yes to no, we should clear out the relevant data, so if spouse goes yes to no, we remove all relevant spouse data (employment history, additional income, benefits). Same should go for stuff like recreational vehicles, vehicles, credit card bills, etc.
> 
> I'm still in the midst of testing, but wanted to drop this in here! I think the spouse removal should be addressed in this set of changes, but the others likely warrant their own tickets to prevent bloat.

The fix is in, it will need some more testing but I think it's doing what we want it to do

**Review by josephrlee — APPROVED** [review]
Content changes look great to me!

**Review by Dr-Pongo — APPROVED** [review]
LGTM! 🚀

**Pull request merged**